app/js/lib/intent-parser.js

```diff
@@ -153,56 +192,58 @@ export default class IntentParser {
   }
 
   /**
-   * Build the `regexps` property as an object mapping locale code to list of
-   * patterns and placeholders pairs.
+   * Build the `patterns` property as an object mapping locale code to list of
+   * patterns.
    */
   [p.init]() {
     Object.keys(PATTERNS).forEach((locale) => {
-      this[p.regexps][locale] = PATTERNS[locale].patterns.map((phrase) =>
-        this[p.buildRegExp](locale, phrase, PATTERNS[locale].placeholders));
+      this[p.patterns][locale] = PATTERNS[locale].patterns.map((phrase) =>
+        this[p.buildPatterns](locale, phrase, PATTERNS[locale].placeholders));
     });
   }
 
-  [p.buildRegExp](locale = 'en', phrase = '', placeholders) {
-    phrase = this[p.normalise](phrase, locale);
-
+  [p.buildPatterns](locale = this.locale, match = '', placeholders) {
+    const phrase = this[p.normalise](match, locale);
     const tokens = this[p.splitOnPlaceholders](phrase);
     const order = {};
     let placeholderIndex = 0;
-    let patterns = tokens.map((token) => {
-      if (token.startsWith('[')) {
-        const placeholder = token
-          .substr(1)
-          // Strip trailing `]`.
-          .replace(new RegExp('\\]$', 'u'), '');
-
-        // The order of the placeholders can be different depending on the
-        // pattern or language. When we parse a string, we need to match the
-        // regexp captured masks to the placeholder given its position.
-        order[placeholder] = placeholderIndex;
-        placeholderIndex++;
-
-        return placeholders[placeholder];
-      }
 
-      if (token === ' ') {
-        return '\\b \\b';
-      }
+    let regexp = tokens
+      .map((token) => {
+        if (token.startsWith('[')) {
+          const placeholder = token
+            .substr(1)
+            // Strip trailing `]`.
+            .replace(new RegExp('\\]$', 'u'), '');
+
+          // The order of the placeholders can be different depending on the
+          // pattern or language. When we parse a string, we need to match the
+          // regexp captured masks to the placeholder given its position.
+          order[placeholder] = placeholderIndex;
+          placeholderIndex++;
+
+          return placeholders[placeholder];
+        }
 
-      // Leading and trailing spaces are changed to word boundary.
-      return this[p.escape](token)
-        .replace(new RegExp('^ ', 'u'), '\\b')
-        .replace(new RegExp(' $', 'u'), '\\b');
-    });
+        if (token === ' ') {
+          return '\\b \\b';
+        }
+
+        // Leading and trailing spaces are changed to word boundary.
+        return this[p.escape](token)
+          .replace(new RegExp('^ ', 'u'), '\\b')
+          .replace(new RegExp(' $', 'u'), '\\b');
+      })
+      .join('');
 
-    patterns = new RegExp(`^${patterns.join('')}$`, 'iu');
+    regexp = new RegExp(`^${regexp}$`, 'iu');
```

Comment:
really minor nit: Is it necessary to rebind `regexp`, could we call it something else with `const`s?